src/Database/Database.php

```diff
@@ -1421,7 +1421,7 @@ public function update($table, $fields, $condition, $old_fields = [])
 			if (is_bool($old_fields)) {
 				if ($do_insert) {
 					$values = array_merge($condition, $fields);
-					return $this->insert($table, $values, $do_insert);
+					return $this->replace($table, $values);
```

Comment:
Shouldn't this call replace the actual `update()` calls when `$old_fields` is set to `true`?

Reply:
`replace` deletes the old entry with the new one. So it can act as a replacement for an `insert`, but not for an `update`.

Reply:
No it doesn't? Replace tries to perform an insert first, then if a unique index is hit, it updates the same row with the provided data. Isn't it exactly what happens with `update($table, $fields, $condition, true)`?

Reply:
See here: https://dev.mysql.com/doc/refman/8.0/en/replace.html
> REPLACE works exactly like INSERT, except that if an old row in the table has the same value as a new row for a PRIMARY KEY or a UNIQUE index, **the old row is deleted before the new row is inserted**.

Reply:
Thanks for the information, I didn't know, however I believe it is irrelevant to the behavior of `update($table, $fields, $condition, true)` that still acts like a `replace()`. I agree it isn't a good replacement for `update()` in general.

Reply:
At some time in the future we can create a function that reflects the direct behaviour of `INSERT ... ON DUPLICATE MODIFY ...`. 

This would contain three parameters:
- tablename
- fields for modify
- additional fields for insert (optional)

With this could handle situations like in the `APContact` table where there is a `created` field that - of course - should only be set upon insert and not upon modify.

But this is something for a different PR.

Reply:
Ah, I see what you mean.